### ISSUE TITLE
Popover/tooltip display issue in WP 5+ when SEO metabox is in the post right sidebar

### ISSUE THREAD

**Issue by dominicht** [sybrew/the-seo-framework#425]
**Describe the bug**
There is an issue displaying the popover/tooltip above the "Doing it right" hints with the Guttenberg editor. If you choose to display the SEO box in the right sidebar, the popover will be cropped and half of its content isn't visible. See screenshot below

**To Reproduce**
1. Drag the SEO box at the bottom of the Guttenberg post editor sidebar
2. Put cursor over the first "Doing it right" hint.

**Expected behavior**
The popover/tooltip should be contained within the sidebar frame.

**Screenshots**
<img width="536" alt="Screen Shot 2019-03-31 at 10 34 45 PM" src="https://user-images.githubusercontent.com/9722069/55300662-80442f80-5406-11e9-9051-4699de5c3284.png">

**Comment by sybrew** [comment]
Since when can you drag legacy meta boxes into the sidebar? I wasn't aware of this.

Looking into this issue, I have found [evidence](https://wordpress.org/support/topic/editor-view-move-gps-layout-metabox-to-sidebar/#post-11131770), but shouldn't you enable this feature for the meta box, as things can and do break? 

Are you using a plugin, theme, or custom script that enforces this behavior?

**Comment by sybrew** [comment]
I wasn't able to move it with my pointer in the browser, but I was able to move it in the code.
```php
add_filter( 'the_seo_framework_metabox_context', function( $context ) { return 'side'; } );
```

With that, I found that nothing broke 🎈 🤷‍♂️ 

This drastically improves the ease of the #387 implementation. I might set the sidebar as standard if `the_seo_framework()->is_gutenberg_page()` yields `true`.

But the overlap is not what we want. I'll add a tooltip boundary in 3.3.0.
You can test this by running this in your browser console:
```js
tsfTT.addBoundary( '.tsf-flex-inside-wrap' );
```

Note that the arrows don't align neatly with the question mark tooltips `[?]`. We'd also want to look at improving this.

**Comment by dominicht** [comment]
I'm not using any plugin or script that affects the Guttenberg editor, but here is what I have installed just in case:
- SEO Framework
- Jetpack/Akismet
- WP Rocket
- Imagify
- Kirki Toolkit
- Mailchimp for Wordpress

I added your filter to move the metabox and gave it a try. I still see the tooltip issue. Then I tried adding:

`tsfTT.addBoundary( '.tsf-flex-inside-wrap' );`

It returned "undefined" in the console but did indeed move the tooltip, but didn't fix the issue. Instead, the tooltips are displayed under the SEO hints instead of above. See screenshot below:

<img width="1792" alt="Screen Shot 2019-04-01 at 10 28 56 AM" src="https://user-images.githubusercontent.com/9722069/55335686-96cfa280-5469-11e9-8ae9-4ef4fa163eb1.png">

As a reference, here is a screenshot of how it was behaving before modifying the boundaries:

<img width="1792" alt="Screen Shot 2019-04-01 at 10 28 27 AM" src="https://user-images.githubusercontent.com/9722069/55335784-c1216000-5469-11e9-9917-af7b2b169bf7.png">

**Comment by sybrew** [comment]
Thanks for the additional details and testing! 

As for `undefined`: That's correct, the method doesn't return a value, it just adds the class to the queryselector you've put in without verification.

I have to add a "overflows left" check for the wrap now, too. This wasn't required earlier (on LTR), because all our tooltips are placed at the right side, without an `overflow:hidden` wrap stopping it.

In hindsight, `.tsf-flex-inside-wrap` isn't the best place, either. We don't want the tooltip to flip while there's still plenty of room left at the top.

**Comment by sybrew** [comment]
Depending on the content width, the tooltip may now grow up to 250px wide (+12px padding) if the tooltip initiator is smaller than 150px, or it will grow up to 250px if the tooltip initiator is larger than 150px. If the content is smaller than any of those values, it'll flow down to the content's size.

If the bounding container is smaller than 250px, it will take the container width, and will try to subtract 12 or 6 pixels of boundary to either side for visual separation.

When it overflows to the left, it'll adjust itself to the right (+6 px separation). If it then detects an overflow to the right, it'll correct itself again to the left (-6px separation). If even then again overflows to the left, it'll finally center itself. This is much like a self-driving car trying to stay between the lines, which is a cool analogy.

With that, the results in the images below look good to me.

## Fit to content size.
![image](https://user-images.githubusercontent.com/7662236/59432257-3359dd80-8de7-11e9-9ab8-1c20fc65703f.png)

## Fit to container size.
![image](https://user-images.githubusercontent.com/7662236/59431693-c560e680-8de5-11e9-97db-d23127e5c0af.png)

![image](https://user-images.githubusercontent.com/7662236/59431719-d578c600-8de5-11e9-82a0-deee361547be.png)

## Fit to initiator size.
![image](https://user-images.githubusercontent.com/7662236/59431703-ceea4e80-8de5-11e9-8c1d-fb69cd452fe6.png)
